x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/selectors.ts

```diff
@@ -73,16 +78,62 @@ export const uiQueryParams: (
   }
 );
 
+export const searchBarQuery: (state: AlertListState) => Query = createSelector(
+  uiQueryParams,
+  ({ query }) => {
+    if (query !== undefined) {
+      return (decode(query) as unknown) as Query;
```

Comment:
this unsafe cast makes me think we might need error handling

Reply:
https://github.com/elastic/endpoint-app-team/issues/265